catalog/forms.py

```diff
@@ -0,0 +1,25 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+import datetime
+
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(
+        help_text=_("Enter a date between now and 4 weeks (default 3).")
+    )
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data["renewal_date"]
```

Comment:
attribute `cleaned_data` có tác dụng gì vậy em?

Reply:
`data = self.cleaned_data["renewal_date"]`:
- *clean_data* có tác dụng giúp "làm sạch" dữ liệu và làm sạch đầu vào có khả năng "không an toàn" bằng cách sử dụng trình xác thực và được chuyển đổi thành loại tiêu chuẩn chính xác cho dữ liệu ạ(trong trường hợp này là đối tượng datetime.datetime trong Python ạ).